### ISSUE TITLE
Publish `.d.ts` type declaration in npm package instead of source `.ts`

### ISSUE THREAD

**Issue by aidenlx** [vidstack/vidstack#638]
I've encountered some typescript check problem when I'm using strict mode in tsconfig. It seems that tsc is reporting strict check errors on `.ts` files published in npm package such as `noImplicitAny`. I initially though it's a configuration problem around `skipLibCheck` but eventually I found this issue which suggests that [publish `.ts` file directly is not good in practice](https://github.com/microsoft/TypeScript/issues/44205#issuecomment-849895090), and `skipLibCheck` explicitly escape `.ts` files inside libs for that reason.

**Comment by mihar-22** [comment]
I see I thought resolved this but it might be because one of the type declaration files is still inside the `src` folder. Specfically [`dom.d.ts`](https://github.com/vidstack/vidstack/blob/main/packages/foundation/src/dom.d.ts). I've moved it out and I'll test if it resolves it. 

It'd be awesome if you could try manually moving `dom.d.ts` in your `node_modules/@vidstack/foundation` folder from `src` to root package roto, updating the sibling `globals.d.ts` file to point to it, and letting me know if it fixes your issues.

**Comment by aidenlx** [comment]
maybe you can try use this to let tsc generate type declaration only: 
```json
{
  "compilerOptions": {
    "emitDeclarationOnly": true,
    "declarationDir": "SOME_TYPE_DIR"
  }
}
```

**Comment by mihar-22** [comment]
That's exactly what we're doing right now.

**Comment by mihar-22** [comment]
Types should be all fixed in latest `1.4.0`!
